tests/integration/test_s3.py

```diff
@@ -154,6 +157,16 @@ def _wait_for_key(self, bucket_name, key_name, extra_params=None,
         for _ in range(min_successes):
             waiter.wait(**params)
 
+    def _check_bucket_versioning(self, bucket, enabled=True):
+        response = self.client.get_bucket_versioning(Bucket=bucket)
+        status = response.get('Status')
+        return status == 'Enabled' if enabled else status != 'Enabled'
+
+    def wait_until_versioning_enabled(self, bucket, min_successes=3):
+        waiter = ConsistencyWaiter()
+        for _ in range(min_successes):
```

Comment:
What's your thoughts on including the min successes in the logic of the `wait` command as well? I suspect we will be relying on min successes quite often.

Reply:
I've been going back and forth on this. I'll incorporate the min success logic into consistency waiter as well.